controllers/vrg_volrep.go

```diff
@@ -1953,41 +1959,46 @@ func (v *VRGInstance) updateExistingPVForSync(pv *corev1.PersistentVolume) error
 	return nil
 }
 
-func (v *VRGInstance) validateExistingObject(obj client.Object) error {
-	switch p := obj.(type) {
-	case *corev1.PersistentVolume:
-		return v.validateExistingPV(p)
-	case *corev1.PersistentVolumeClaim:
-		return v.validateExistingPVC(p)
-	default:
-		return fmt.Errorf("unknown object %v", obj)
-	}
-}
-
 func (v *VRGInstance) validateExistingPV(pv *corev1.PersistentVolume) error {
 	existingPV := &corev1.PersistentVolume{}
+	log := v.log.WithValues("PV", existingPV.GetName())
 
 	err := v.reconciler.Get(v.ctx, types.NamespacedName{Name: pv.Name}, existingPV)
 	if err != nil {
 		return fmt.Errorf("failed to get PV (%s) (%w)", existingPV.GetName(), err)
 	}
 
-	// If the PV is bound and matches with the PV we would have restored then return now
-	if v.pvMatches(existingPV, pv) {
-		v.log.Info("Existing PV matches and is bound to the same claim", "PV", existingPV.GetName())
+	if existingPV.Status.Phase == corev1.VolumeBound {
+		var pvc corev1.PersistentVolumeClaim
 
-		return nil
-	}
+		pvcNamespacedName := types.NamespacedName{Name: pv.Spec.ClaimRef.Name, Namespace: pv.Spec.ClaimRef.Namespace}
+
+		err := v.reconciler.Get(v.ctx, pvcNamespacedName, &pvc)
+		if err != nil {
+			return fmt.Errorf("failed to get PV %s claim %s: %w", existingPV.GetName(), pvcNamespacedName.String(), err)
```

Comment:
(nits) Thinking if the error message can be changed like so:

"found bound PV %s to claim %s but unable to validate claim exists: %w"

and something similar in 1982 stating deleted claim (maybe just adding "bound" word between "existing bound PV..."

Reason being, we expect the situation to be rectified as the PV should reach the Released state at some point in time. So failing to get a claim that does not exist is fine. (and that this message may/will reach the VRG conditions)

Reply:
Fixed